conf/logging.go

```diff
@@ -2,59 +2,54 @@ package conf
 
 import (
 	"os"
+	"strings"
 	"time"
 
-	"github.com/sirupsen/logrus"
+	"github.com/rs/zerolog/log"
+
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/pkgerrors"
 )
 
 type LoggingConfig struct {
-	Level            string                 `mapstructure:"log_level" json:"log_level"`
+	Level            string                 `mapstructure:"log_level" json:"log_level"  default:"console"` // defaults to std log - developer friendly
 	File             string                 `mapstructure:"log_file" json:"log_file"`
 	DisableColors    bool                   `mapstructure:"disable_colors" split_words:"true" json:"disable_colors"`
 	QuoteEmptyFields bool                   `mapstructure:"quote_empty_fields" split_words:"true" json:"quote_empty_fields"`
 	TSFormat         string                 `mapstructure:"ts_format" json:"ts_format"`
 	Fields           map[string]interface{} `mapstructure:"fields" json:"fields"`
+	Format           string                 `mapstructure:"format" json:"format" default:"stdout"`
 }
 
-func ConfigureLogging(config *LoggingConfig) (*logrus.Entry, error) {
-	logger := logrus.New()
-
-	tsFormat := time.RFC3339Nano
-	if config.TSFormat != "" {
-		tsFormat = config.TSFormat
+// trim full path. output in the form directory/file.go.
+func consoleFormatCaller(i interface{}) string {
+	var c string
+	if cc, ok := i.(string); ok {
+		c = cc
 	}
-	// always use the full timestamp
-	logger.SetFormatter(&logrus.TextFormatter{
-		FullTimestamp:    true,
-		DisableTimestamp: false,
-		TimestampFormat:  tsFormat,
-		DisableColors:    config.DisableColors,
-		QuoteEmptyFields: config.QuoteEmptyFields,
-	})
-
-	// use a file if you want
-	if config.File != "" {
-		f, errOpen := os.OpenFile(config.File, os.O_RDWR|os.O_APPEND|os.O_CREATE, 0664)
-		if errOpen != nil {
-			return nil, errOpen
+	if len(c) > 0 {
+		l := strings.Split(c, "/")
+		if len(l) == 1 {
+			return l[0]
 		}
-		logger.SetOutput(f)
-		logger.Infof("Set output file to %s", config.File)
+		return l[len(l)-2] + "/" + l[len(l)-1]
```

Comment:
do we need to check length here as well?

Reply:
length cases

0 - it cannot have 0 length
1 - it is handled
2 - works fine with l-2 (0)
3 and onwards - works same as above